scripts/fast-reboot

```diff
@@ -806,6 +806,17 @@ fi
 
 # Reboot: explicitly call Linux native reboot under sbin
 debug "Rebooting with ${REBOOT_METHOD} to ${NEXT_SONIC_IMAGE} ..."
+
+LOGS_ON_TMPFS=0
+df --output=fstype /var/log* | grep -c 'tmpfs' || LOGS_ON_TMPFS=$?
```

Comment:
grep -c will always return 0, what you want is the output of grep -c

LOGS_ON_TMPFS=$(df --output=fstype /var/log* | grep -c 'tmpfs')

Reply:
And you want to check the count is greater than 0.